### ISSUE TITLE
Migrations: Provider rendered-annotations in code

### ISSUE THREAD

**Issue by bricelam** [aspnet/EntityFramework#6546]
This was part of #1825, but I'm splitting it out so it can be prioritized separately.

Currently, annotations on migration operations (and in the model snapshot) are rendered generally by the code generator:
```C#
migrationBuilder.AddColumn<int>(
    name: "Id",
    table: "Person")
    .Annotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn);
```

Similar to how relational annotations are handled when scaffolding the model snapshot, providers should be given an opportunity to render these annotations to provider-specific extension methods. For example:

```C#
migrationBuilder.AddColumn<int>(
    name: "Id",
    table: "Person")
    .UseSqlServerIdentityColumn();
```

If a provider cannot render the annotation for the current language, the original syntax will be used as a fallback.

**Comment by roji** [comment]
Whatever mechanism the provider will have to specify its own rendering to annotations in migrations, note the importance of passing it the old annotation value in addition to the new. This is necessary to support more advanced scenario where structured data is represented in the annotation value, e.g. a list of PostgreSQL extensions.

**Comment by bricelam** [comment]
Yes, AlterDatabase, AlterTable, and AlterColumn have both the new and old annotation values. Drop operations will also have any old annotations provided by `IMigrationsAnnotationProvider.ForRemove()`.

**Comment by roji** [comment]
Wasn't this done in #21329 (there was really a tracking issue aside from #16922)?

**Comment by bricelam** [comment]
No, this is about annotations inside the Up and Down methods--not the snapshot.
